### PR TITLE
Tl/support auth persist

### PR DESCRIPTION
**Pull request by tungsheng**
# Summary
This branch implements features includes:
- wehdfs support
- website login support
- print mode
- pdf header/footer

# Screenshots
N/A

# Tests implemented
N/A

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -11,6 +11,21 @@
...
+export SIGN_IN_SELECTOR=xxx
+export AUTH_PATH=xxx/xxx
+export AUTH_USERNAME=xxx
+export AUTH_PASSWORD=xxx
+export ALLOW_UNKNOWN_BODY=true
+export HDFS_USER=root
+export HDFS_HOST=localhost
+export HDFS_PORT=50070
+export HDFS_PATH=/webhdfs/v1
+export IGNORE_HTTPS_ERRORS=true
```
Comment (visthakk) [inline .env.sample]: Thanks. This will help for templating.

File: src/core/render-core.js [hunk 2]
```diff
@@ -150,6 +165,10 @@ async function render(_opts = {}) {
 
       data = await page.screenshot(screenshotOpts);
     }
+
+    // Write data to file
+    logger.info('[renderCore] write to file, jobId = ', opts.jobId);
```
Comment (visthakk) [inline src/core/render-core.js:170]: @tungsheng : Do we have a way to direct logging to a file?

File: src/core/render-core.js [hunk 3]
```diff
@@ -150,6 +165,10 @@ async function render(_opts = {}) {
...
+    await writeToFile(`/home/pdf/${opts.jobId}.pdf`, data);
```
Comment (visthakk) [inline src/core/render-core.js:171]: Is this the path in hdfs? We should parameterize top level dir.

File: src/core/render-core.js (hunk 2 above)
Comment (tungsheng) [reply src/core/render-core.js:170]: @visthakk I think so, should be able to save it to a text file in HDFS or other options. When do you think we will need to implement this logging feature?

File: src/core/render-core.js (hunk 3 above)
Comment (tungsheng) [reply src/core/render-core.js:171]: Yes, this is a temporary location for pdf file. Do we have a permanent location? Or do we want to pass in the location in post body?

Comment (ali-tetration) [reply src/core/render-core.js:171]: I agree that the location on HDFS should be parameterized and managed by ansible. This way we can make sure the directory with proper permissions is created before running the service. The path can then be returned when done writing to HDFS, but is not required to be provided in the post body

File: .env.sample [hunk 4]
```diff
@@ -11,6 +11,21 @@
...
+export PASSWD_SELECTOR=xxx
+export SIGN_IN_SELECTOR=xxx
+export AUTH_PATH=xxx/xxx
+export AUTH_USERNAME=xxx
+export AUTH_PASSWORD=xxx
+export ALLOW_UNKNOWN_BODY=true
+export HDFS_USER=root
+export HDFS_HOST=localhost
+export HDFS_PORT=50070
+export HDFS_PATH=/webhdfs/v1
```
Comment (ali-tetration) [inline .env.sample]: Maybe call it `WEBHDFS_URL` or something like that to avoid confusion with file path

File: src/core/render-core.js [hunk 5]
```diff
@@ -38,13 +50,15 @@ async function render(_opts = {}) {
...
-    headless: !config.DEBUG_MODE,
-    ignoreHTTPSErrors: opts.ignoreHttpsErrors,
+    headless: opts.headless || !config.DEBUG_MODE,
+    ignoreHTTPSErrors: opts.ignoreHttpsErrors || config.IGNORE_HTTPS_ERRORS,
     args: ['--disable-gpu', '--no-sandbox', '--disable-setuid-sandbox'],
     sloMo: config.DEBUG_MODE ? 250 : undefined,
   });
   const page = await browser.newPage();
 
+  if (opts.needAuth) await getAuth(page, opts.username, opts.passwd);
```
Comment (ali-tetration) [inline src/core/render-core.js:60]: We need a way to not have to navigate to login page and enter user/pass every time we need to print a page. Could we provision a way to stay logged in for certain amount of time for a special user? I am ok with this for the MVP but we really need a better solution before shipping this

cc. @jpang-h4

File: src/core/render-core.js (hunk 2 above)
Comment (visthakk) [reply src/core/render-core.js:170]: Log rotation added in deploy script.

**Pull request merged**